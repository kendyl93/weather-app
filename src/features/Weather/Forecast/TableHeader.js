import React from "react"
import { makeStyles } from "@material-ui/styles"
import Typography from "@material-ui/core/Typography"

const useStyles = makeStyles((theme) => ({
  tableHeader: {
    display: "grid",
    gridTemplateColumns: "auto 50px 50px",
    alignItems: "center",
    textAlign: "center",

    "& p": {
      fontWeight: 700,
    },
  },
}))

const TableHeader = () => {
  const classes = useStyles()
  return (
    <div className={classes.tableHeader} data-testid="table-header">
      <div></div>
      <Typography>Low</Typography>
      <Typography>Hight</Typography>
    </div>
  )
}

export default TableHeader
