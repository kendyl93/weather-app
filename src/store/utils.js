export const LOADING_STARTED = "LOADING_STARTED"
export const LOADING_FINISHED = "LOADING_FINISHED"

export const setLoadingStarted = () => ({
  type: LOADING_STARTED,
})

export const setLoadingFinished = () => ({
  type: LOADING_FINISHED,
})

const withLoading = async (dispatch, callback) => {
  dispatch(setLoadingStarted())
  await callback()
  dispatch(setLoadingFinished())
}

export default withLoading
