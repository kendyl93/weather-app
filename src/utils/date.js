const ONE_SECOND = 1000

const fromSecondsToMiliseconds = (seconds) => seconds * ONE_SECOND

export const formatDate = (timestamp, offset = 0) => {
  const dateTimeInMiliseconds = fromSecondsToMiliseconds(timestamp)
  const offsetInMiliseconds = fromSecondsToMiliseconds(offset)
  const date = new Date(dateTimeInMiliseconds + offsetInMiliseconds).toISOString()
  const options = {
    weekday: "short",
    year: "numeric",
    month: "short",
    day: "2-digit",
  }

  const formattedDate = new Intl.DateTimeFormat("en-US", options).format(
    new Date(date)
  )
  return formattedDate
}
