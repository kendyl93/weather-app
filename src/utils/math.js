import { MAP_ZOOM } from "../constants"

export const longitude2tile = (lon) => {
  if (typeof lon !== "number") {
    return 0
  }

  return Math.floor(((lon + 180) / 360) * Math.pow(2, MAP_ZOOM))
}

export const latitude2tile = (lat) => {
  if (typeof lat !== "number") {
    return 0
  }

  return Math.floor(
    ((1 -
      Math.log(
        Math.tan((lat * Math.PI) / 180) + 1 / Math.cos((lat * Math.PI) / 180)
      ) /
        Math.PI) /
      2) *
      Math.pow(2, MAP_ZOOM)
  )
}
