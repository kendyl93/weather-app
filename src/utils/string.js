export const celcius = (temperature) => `${Math.round(temperature)}\xB0C`
