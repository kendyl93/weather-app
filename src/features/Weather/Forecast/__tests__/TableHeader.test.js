import React from "react"
import { cleanup, render } from "../../../../tests/utils"
import TableHeader from "../TableHeader"

afterEach(cleanup)

describe("TableHeader snapshot test", () => {
  test("should match base snapshot", () => {
    const { asFragment } = render(<TableHeader />)

    expect(asFragment(<TableHeader />)).toMatchSnapshot()
  })
})
