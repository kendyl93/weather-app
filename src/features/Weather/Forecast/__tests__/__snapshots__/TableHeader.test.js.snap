// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableHeader snapshot test should match base snapshot 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-tableHeader-1"
    data-testid="table-header"
  >
    <div />
    <p
      class="MuiTypography-root MuiTypography-body1"
    >
      Low
    </p>
    <p
      class="MuiTypography-root MuiTypography-body1"
    >
      Hight
    </p>
  </div>
</DocumentFragment>
`;
